### PR TITLE
Convert controller job and namespace files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

File: pkg/controller/job/job_controller.go [hunk 1]
```diff
@@ -858,7 +858,7 @@ func (jm *Controller) manageJob(job *batch.Job, activePods []*v1.Pod, succeeded
 					if err != nil {
 						defer utilruntime.HandleError(err)
 						// Decrement the expected number of creates because the informer won't observe this pod
-						klog.V(2).Infof("Failed creation, decrementing expectations for job %q/%q", job.Namespace, job.Name)
+						klog.V(2).InfoS("Failed creation, decrementing expectations for Job", "jobNamespace", job.Namespace, "job", job.Name)
```
Comment (yangjunmyfm192085) [inline pkg/controller/job/job_controller.go:861]: Maybe consider using KObj(obj ObjectMeta)  or KRef(namespace, name string)?

File: pkg/controller/job/job_controller.go [hunk 2]
```diff
@@ -869,7 +869,7 @@ func (jm *Controller) manageJob(job *batch.Job, activePods []*v1.Pod, succeeded
 			// any skipped pods that we never attempted to start shouldn't be expected.
 			skippedPods := diff - batchSize
 			if errorCount < len(errCh) && skippedPods > 0 {
-				klog.V(2).Infof("Slow-start failure. Skipping creation of %d pods, decrementing expectations for job %q/%q", skippedPods, job.Namespace, job.Name)
+				klog.V(2).InfoS("Slow-start failure. Skipping creation of Pods, decrementing expectations for Job", "skippedPods", skippedPods, "jobNamespace", job.Namespace, "job", job.Name)
```
Comment (yangjunmyfm192085) [inline pkg/controller/job/job_controller.go:872]: Maybe consider using KObj(obj ObjectMeta) or KRef(namespace, name string)?

File: pkg/controller/job/job_controller.go (hunk 1 above)
Comment (scottilee) [reply pkg/controller/job/job_controller.go:861]: Fixed.

File: pkg/controller/job/job_controller.go (hunk 2 above)
Comment (scottilee) [reply pkg/controller/job/job_controller.go:872]: Fixed.

File: pkg/controller/namespace/deletion/namespaced_resources_deleter.go [hunk 3]
```diff
@@ -551,14 +551,14 @@ func (d *namespacedResourcesDeleter) deleteAllContent(ns *v1.Namespace) (int64,
 	}
 
 	// if len(errs)==0, NewAggregate returns nil.
-	klog.V(4).Infof("namespace controller - deleteAllContent - namespace: %s, estimate: %v, errors: %v", namespace, estimate, utilerrors.NewAggregate(errs))
+	klog.V(4).InfoS("Namespace controller - deleteAllContent", "namespace", namespace, "estimate", estimate, "err", utilerrors.NewAggregate(errs))
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controller/namespace/deletion/namespaced_resources_deleter.go:554]: Hey good question. This section of https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments says to use "err". Let me know if you think if should be "errors" and I can change it.

**Comment by scottilee** [comment]
@yangjunmyfm192085 and @cici37 Thanks for you reviews! I've addressed your comments. Let me know if there's anything else I should fix.

**Comment by cici37** [comment]
@scottilee You may want to update `kubernetes/pkg/controller/namespace/deletion/namespaced_resources_deleter.go:164` as well.

File: pkg/controller/namespace/deletion/namespaced_resources_deleter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cici37) [reply pkg/controller/namespace/deletion/namespaced_resources_deleter.go:554]: My previous concern is more about if we should indicate there would be an error list to be expected. Thanks for sharing the guidance. "err" should also work :)

**Comment by scottilee** [comment]
> @scottilee You may want to update `kubernetes/pkg/controller/namespace/deletion/namespaced_resources_deleter.go:164` as well.

Fixed!

**Comment by scottilee** [comment]
Rebased.

**Pull request closed without merging**